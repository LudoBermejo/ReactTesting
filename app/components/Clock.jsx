import React from 'react';

String.prototype.padZero = function (len, c) {
  var s = '',
    c = c || '0',
    len = (len || 2) - this.length;
  while (s.length < len) s += c;
  return s + this;
};

Number.prototype.padZero = function (len, c) {
  return String(this).padZero(len, c);
};

export default class Clock extends React.Component {
  formatSeconds(totalSeconds) {
    if (!totalSeconds || typeof (totalSeconds) !== 'number') return false;
    const seconds = (totalSeconds % 60).padZero(2);
    const minutes = Math.floor(totalSeconds / 60).padZero(2);
    return `${minutes}:${seconds}`;
  }
  render() {
    return (
      <div>This is my clock</div>
    );
  }
}
