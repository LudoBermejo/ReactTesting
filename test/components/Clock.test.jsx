import React from 'react';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';

import Clock from 'Clock';

describe('Clock component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Clock />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renderTime', () => {
    let clock;
    beforeEach(() => {
      clock = TestUtils.renderIntoDocument(<Clock />);
    });

    it('returns false if empty string is sent', () => {
      expect(clock.formatSeconds('')).toBeFalsy();
    });

    it('returns false if string is sent', () => {
      expect(clock.formatSeconds('demo')).toBeFalsy();
    });

    it('returns false if array is sent', () => {
      expect(clock.formatSeconds([])).toBeFalsy();
    });

    it('returns valid data if less that 10 seconds is sent', () => {
      expect(clock.formatSeconds(9)).toBe('00:09');
    })

    it('returns valid data if less that 60 seconds is sent', () => {
      expect(clock.formatSeconds(50)).toBe('00:50');
    })

    it('returns valid data if more that 60 seconds is sent', () => {
      expect(clock.formatSeconds(140)).toBe('02:20');
    })

  });
});
