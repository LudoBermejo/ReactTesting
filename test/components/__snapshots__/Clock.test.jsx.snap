exports[`Clock component renders correctly 1`] = `
<div>
  This is my clock
</div>
`;
